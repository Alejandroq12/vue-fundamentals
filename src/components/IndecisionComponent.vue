<template>
  <img src="https://via.placeholder.com/250" alt="bg">
  <div class="bg-dark"></div>  
  <div class="indecision-container">
      <input type="text" placeholder="Hazme una pregunta" v-model="question">
      <p>Recuerda terminar con un signo de interrogacion (?)</p>
      <div>
          <h2 :question="question">{{ question }}</h2>
          <h1>Si, No, .... pensando</h1>
      </div>
  </div>
</template>

<script>
export default {
  data: () => ({
    question: '',
  })
};
</script>

<style scoped>

    img, .bg-dark {
        height: 100vh;
        left: 0px;
        max-height: 100%;
        max-width: 100%;
        position: fixed;
        top: 0px;
        width: 100vw;
    }

    .bg-dark {
        background-color: rgba(0, 0, 0, 0.4);
    }

    .indecision-container {
        position: relative;
        z-index: 99;
    }
    
    input {
        width: 250px;
        padding: 10px 15px;
        border-radius: 5px;
        border: none;
    }
    input:focus {
        outline: none;
    }

    p {
        color: white;
        font-size: 20px;
        margin-top: 0px;
    }

    h1, h2 {
        color: white;
    }
    
    h2 {
        margin-top: 150px;
    }

</style>