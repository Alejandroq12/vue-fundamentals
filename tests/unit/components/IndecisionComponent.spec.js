import { shallowMount } from '@vue/test-utils';
import IndecisionComponent from '../../../src/components/IndecisionComponent.vue';

describe('IndecisionComponent', () => {
  let wrapper;
  let clgSpy;

  beforeEach(() => {
    wrapper = shallowMount(IndecisionComponent);

    clgSpy = jest.spyOn(console, 'log');
  });
  test('Debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Should not dispatch when typing on input console.log()', async () => {
    const getAnswerSpy = jest.spyOn(wrapper.vm, 'getAnswer')
    const input = wrapper.find('input');
    await input.setValue('Hola mundo');

    expect(clgSpy).toHaveBeenCalledTimes(1);
    expect(getAnswerSpy).not.toHaveBeenCalled();

  });

  test('Question mark "?" must dispatch fecth', () => {

  })

  test('getAnswer tests', () => {

  })

  test('getAnswer test - API failure', () => {

  })
});
