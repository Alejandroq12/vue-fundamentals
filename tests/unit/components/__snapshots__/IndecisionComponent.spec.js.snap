// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndecisionComponent Debe de hacer match con el snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogacion (?)</p>
  <!--v-if-->
</div>
`;
